base/special/gamma.jl

```diff
@@ -482,3 +482,7 @@ eta(x::Integer) = eta(Float64(x))
 eta(x::Real)    = oftype(float(x),eta(Float64(x)))
 eta(z::Complex) = oftype(float(z),eta(Complex128(z)))
 @vectorize_1arg Number eta
+
+Base.binomial{S<:Number,T<:Number}(n::S, k::T) = ( (isinteger(n) && isinteger(k))
```

Comment:
If you know when it's appropriate to overflow I think it's better to do so rather than throw an `InexactError` in `round(Int64, n)`. This also needs quite a few more tests.